### DISCUSSION TITLE
超越Java、C語言   Python優點登上最受歡迎的程式語言

### DISCUSSION THREAD

**Discussion by 239216** [1101-GS4719-C/Python-Community-GS4719-C#91] [General]
依照近期公布的TIOBE編程社群指數顯示，Python已經超越Java、JavaScript、C語言成為目前最受歡迎的程式語言。
Python在1991年由荷蘭工程師Guido van Rossum創造，強調程式碼可讀性與簡潔語法，相比C語言、Java等程式語言能以更少編碼內容撰寫，並且讓程式結構更簡單，並且幾乎能在所有作業系統環境下執行，因此廣受許多程式編碼工作人員喜愛。
近年有越來越多人工智慧、機器學習，以及數據分析應用需求，因此也讓能使編碼內容精簡呈現的Python變成許多程式設計人員使用語言，光在去年就已經增加220萬名Python開發人員。
https://udn.com/news/story/7086/5820271

**Comment by 109802503gloria** [comment]
剛剛也看到這則新聞，想分享但同學捷足先登了XD

**Comment by jerry850313** [comment]
我覺得Python會起來的主要原因是相對其他語言而言，比較容易學習，所以入門門檻降低，
跟C剛出來的環境類似。有趣的是除了有C語言還有A語言、B語言和D語言。
1951 - Regional Assembly Language
1952 - Autocode
1954 - FORTRAN
1954 - IPL (LISP的先驅)
1955 - FLOW-MATIC (COBOL的先驅)
1957 - COMTRAN (COBOL的先驅)
1958 - LISP
1958 - ALGOL 58
1959 - FACT (COBOL的先驅)
1959 - COBOL
1962 - APL
1962 - Simula
1962 - SNOBOL
1963 - CPL (C的先驅)
1964 - BASIC
1964 - PL/I
1967 - BCPL (C的先驅)
1968 - Logo
1970 - Pascal
1970 - Forth
1972 - C
1972 - Smalltalk
1972 - Prolog
1973 - ML
1975 - Scheme
1980 - Ada
1983 - C++ (就像有類別的C)
1984 - Common Lisp
1985 - Eiffel
1986 - Erlang
1987 - Perl
1988 - Tcl
1989 - FL (Backus)
1990 - Haskell
1991 - Python
1991 - Visual Basic
1993 - Ruby
1993 - Lua
1994 - CLOS (part of ANSI Common Lisp)
1995 - Java
1995 - Delphi (Object Pascal)
1995 - JavaScript
1995 - PHP
1997 - REBOL
1999 - D
2001 - C#
2001 - Visual Basic .NET
2002 - F#
2003 - Scala
2003 - Factor
2006 - Windows PowerShell
2007 - Clojure
2009 - Go
2014 - Swift (程式語言)
以上是程式語言開發簡史(data from [wiki](https://zh.wikipedia.org/wiki/%E7%A8%8B%E5%BC%8F%E8%AA%9E%E8%A8%80%E6%AD%B7%E5%8F%B2))

**Reply by jerry850313** [reply]
BTW
如果沒有程式語言，現在我們還在用[這個](https://zh.wikipedia.org/wiki/%E6%89%93%E5%AD%94%E5%8D%A1)
